core/src/main/scala/kafka/cluster/Partition.scala

```diff
@@ -1215,8 +1215,14 @@ class Partition(val topicPartition: TopicPartition,
     }
   }
 
-  def appendRecordsToLeader(records: MemoryRecords, origin: AppendOrigin, requiredAcks: Int,
-                            requestLocal: RequestLocal, verificationGuard: VerificationGuard = VerificationGuard.SENTINEL): LogAppendInfo = {
+  def appendRecordsToLeader(
+    records: MemoryRecords,
+    origin: AppendOrigin,
+    requiredAcks: Int,
+    requestLocal: RequestLocal,
+    verificationGuard: VerificationGuard = VerificationGuard.SENTINEL,
+    transactionVersion: Short = 0
```

Comment:
Can we make all default arguments to pass -1 (add a constant in `TransactionVersion`, say `static final short TV_UNKNOWN = -1`) and then in `appendEndTxnMarker` throw invalid argument exception?  Otherwise it's too easy to accidentally use a method with default and not pass proper version (which would be then treated as TV_0).